lib/statesman/adapters/active_record.rb

```diff
@@ -96,21 +120,58 @@ def transitions_for_parent
         parent_model.send(@association_name)
       end
 
-      def unset_old_most_recent
-        most_recent = transitions_for_parent.where(most_recent: true)
+      # Sets the given transition most_recent = t while unsetting the most_recent of any
+      # previous transitions.
+      def update_most_recents(most_recent_id)
+        transitions = transitions_for_parent
+        last_or_current = transitions.where(id: most_recent_id).or(
```

Comment:
Equivalent SQL should be:

```
transitions.where("#{transition_class.table_name}.id = #{most_recent_id} OR #{transition_class.table_name}.most_recent = #{::ActiveRecord::Base.connection.quote(true)}")
```

The interpolation should be safe, too, since everything's from statesman internals.

We use a similar approach in `Statesman::Adapters::ActiveRecordQueries`.

Can PR it if you're 👍 on the approach.

Reply:
Absolutely- please do just add it to this PR!

Slightly concerned about `table_name` (if the table name is hyphenated, then I think this will break depending on what database you use?) but if we use it elsewhere then it's unlikely to be something people rely on.

Reply:
Think we use it elsewhere, but no reason not to quote the shit out of it 🤷‍♂

Reply:
Better to use AREL in this case, to be more safe and database agnostic.